### ISSUE TITLE
Brewfile formulae

### ISSUE THREAD

**Issue by ivankatliarchuk** [tmux-python/tmuxp#593]
### Step 1: Provide a summary of your problem
  * For general technical questions, problems or feature requests related to the code **in this repository** file an issue.
Installation over Brewfile
Macos and Linux users can have it installed with brewfile
This should simplify documentation and installation since it will be bundled

### Step 2: Provide text details
  * Python version
  * system PATH
  * tmux version
  * tmuxp version
  * tmux path
  * tmuxp path,
  * libtmux version
  * shell
  * output of `tmux list-sessions`, `tmux list-windows`, `tmux list-panes`
  * output of `tmux show-options -g`, `tmux show-window-options -g`
  * output of `tmuxp freeze <SESSION_NAME>`
This is an improvement

### Step 3: Describe your environment
  * Architecture: _____
  * OS version: MacOS, Debian

#### Steps to reproduce:
  1. pip is fine
  2. Brew is cool

#### Observed Results:
  * What happened?  This could be a description, log output, etc.

#### Expected Results:
  * What did you expect to happen?

#### Relevant Code:
  ```
  // TODO(you): paste here tmuxp configuration (YAML / JSON) you are having issues with.

**Comment by jvcarli** [comment]
Hi, I've wrote a [custom](https://github.com/jvcarli/homebrew-tap/blob/main/tmuxp.rb) brew tap for tmuxp. I've been using it for the last month and had no problems so far, you can test it if you want.

If everything is correct within the formula I can submit a pull request.

**Comment by tony** [comment]
In order to have an official brew formulae, what would we need to do?

@jvcarli Is this something that could be added as an official formula if you PR'd it there?

**Comment by jvcarli** [comment]
@tony Yes, this could be added as an official formula. This would be the best approach actually. We would need to send a pull request to [homebrew-core](https://github.com/Homebrew/homebrew-core). 

I hadn't done it yet because I've been testing the tap for the last month and I've thought the formula would be best debated here before submitting it to homebrew-core.

As far as I can tell everything is working but it would be nice to add:

- Tests for the formula. Homebrew has [support](https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula) for them. I think it would be relevant to add some besides a simple `tmuxp --version`
- Add shell completion (bash, fish, zsh) install together with tmuxp package. Formulas have native support for this too.

These aren't mandatory for the formula to be accepted, the mandatory parts were already done. I've audited the formula using `brew audit --strict tmuxp` and fixed all the errors I found

**Comment by tony** [comment]
@jvcarli A test would be good, as would shell completions - which I haven't used in some time (i haven't looked at shell completions since we moved to click years ago), I should check on them.

Good to know that `brew audit --strict tmuxp` passes! You have permission to send the formula in (not that you need it 😆).  We can always see if that gets merged, and follow up with completions / tests after.

If it's not accepted, we can also look at adding tests and completions to the formula to see if that eases the way in.

I am a bit strapped on this since I don't have a mac machine (though I think homebrew works on linux now?)

**Comment by jvcarli** [comment]
Homebrew works on Linux too although I never use with it and forgot about it.

I had some time and tested the formula on a [Debian 10](https://app.vagrantup.com/debian/boxes/buster64/versions/10.20210409.1) vagrant box, everything seems to be working as expected. Feel free to test from your side.

I will submit a pull request to [homebrew-core](https://github.com/Homebrew/homebrew-core), then we can track any issues from there.

**Comment by askedrelic** [comment]
I'll contribute: I installed the tmuxp.rb homebrew formula correctly on a standard homebrew/OSX 10.15 setup, so +1

I've been using pipx to install tmuxp, maybe it's easier to consolidate on homebrew!

**Comment by tony** [comment]
@askedrelic brilliant!  

@jvcarli do report back here when you have it published, we'll keep an eye on it and i'll back you up if you need any assistance / additional help. other's can join in here too!

**Comment by tony** [comment]
@jvcarli I left a comment at to confirm if `tmuxp debug-info` works: https://github.com/Homebrew/homebrew-core/pull/79011#discussion_r648245701


I think we can change the test to use `tmuxp debug-info` and verify libtmux, tmux version, etc (that it's returned, or if the testsuite makes it available the version itself).  

Examples of a test on another python cli app:
- https://github.com/Homebrew/homebrew-core/blob/3f10e11/Formula/isort.rb#L28
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/black.rb#L113
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/ipython.rb#L175
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/poetry.rb#L224
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/ptpython.rb#L105
- https://github.com/Homebrew/homebrew-core/blob/67a72b0/Formula/virtualenv.rb#L44
- https://github.com/Homebrew/homebrew-core/blob/216ea6a/Formula/pyenv-virtualenvwrapper.rb#L18

**Comment by tony** [comment]
@jvcarli We can go ahead and just to `tmuxp debug-info` for now since I think that's the best of all worlds.  As to what you'd like to assert I will leave it to your discretion. if you need more from me let me know

**Comment by jvcarli** [comment]
@askedrelic You can uninstall tmuxp and remove my tap because tmuxp is in homebrew-core now! `brew install tmuxp` is  working as expected. Thank you for testing and thank you @tony for the support during the process.

**Comment by tony** [comment]
@jvcarli Awesome, way to go @jvcarli!

https://github.com/Homebrew/homebrew-core/commit/aff29be268687da76f0a01a3740ecb01f0e8e89f
